jetty-client/src/main/java/org/eclipse/jetty/client/HttpRequest.java

```diff
@@ -677,14 +678,31 @@ public ContentResponse send() throws InterruptedException, TimeoutException, Exe
 
         try
         {
-            long timeout = getTimeout();
-            if (timeout <= 0)
-                return listener.get();
+            return listener.get();
+        }
+        catch (ExecutionException x)
+        {            
+            // Previously this method used a timed get on the future, which was in a race
+            // with the timeouts implemented in HttpDestination and HttpConnection. The change to
+            // make those timeouts relative to the timestamp taken in sent() has made that race
+            // less certain, so a timeout could be either a TimeoutException from the get() or
+            // a ExecutionException(TimeoutException) from the HttpDestination/HttpConnection.
+            // We now do not do a timed get and just rely on the HttpDestination/HttpConnection
+            // timeouts.   This has the affect of changing this method from mostly throwing a
+            // TimeoutException to always throwing a ExecutionException(TimeoutException).
+            // Thus for backwards compatibility we unwrap the timeout exception here
+            if (x.getCause() instanceof TimeoutException)
+            {
+                TimeoutException t = (TimeoutException) (x.getCause());
+                abort(t);
+                throw t;
+            }
 
-            return listener.get(timeout, TimeUnit.MILLISECONDS);
```

Comment:
@sbordet I think this was a race that was causing a lot of the HttpClient test instability.  By using the request timeout as the timeout on the `future.get`, we have two timeouts racing each other, but with different results.  One throws a TimeoutException and the other throws an ExecutionException wrapping a TimeoutException.     I don't think we need both, so I have removed the `future.get` timeout, as it is needless extra scheduling.   But I am also now unwrapping the exectionexception to try to keep the API behaviour the same... see comment in the new code.